x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/v2/AbstractFieldProducer.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.rollup.v2;
+
+public abstract class AbstractFieldProducer<T> implements Collectable<T> {
```

Comment:
Something like this deserves javadoc I think. Also, can it be package private? It looks like something specific to rollup v2 and package private just gives the extra context that "this is contained".